src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/lib/get_matched_indices.js

```diff
@@ -26,9 +26,21 @@ export function getMatchedIndices(
     if (name === query) {
       return true;
     }
-    if (query.endsWith('*') && name.indexOf(query.substring(0, query.length - 1)) === 0) {
-      return true;
+
+    const regexQuery = query.startsWith('*')
```

Comment:
How about extracting this logic into a helper function, and adding a couple light tests for it? I think it would improve readability of this code.

Reply:
Sounds like the right move!